### PR TITLE
Update RecordFileParser.loadRecordFile() to match design

### LINKED ISSUE #568: Move filesystem-dependency from RecordFileParser to (new) RecordFileReader class
Right now, `RecordFileParser` class has two concerns:
- periodically - list files, read files (depends on 'where' of stream files)
- parse stream files.

We want to spit it into
- `RecordFileReader`: handles listing/reading of files. Use `FileWatcher` like in `BalanceFileParser`. Share code (if possible)
- `RecordFileParser`: Only cares about parsing of stream file (given as `InputStream` to not depend on filesystem aspects)

### PR DESCRIPTION
**Pull request by apeksharma**
**Detailed description**:
Most RFP tests are at parse() level, so not much test changes.
When we separate fs stuff from RFP, tests will be updated to test just loadRecordFile().

**Which issue(s) this PR fixes**:
Partially addresses #568. All FS code in 2 functions now, can be moved to RecordFileReader later.

**Special notes for your reviewer**:
Address the comment here: https://github.com/hashgraph/hedera-mirror-node/pull/585#discussion_r389216145

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/RecordFileParserTest.java [hunk 1]
```diff
@@ -165,7 +165,9 @@ void invalidFile() throws Exception {
         recordFileParser.parse();
 
         // then
-        assertNoneProcessed();
+        assertParsedFiles();
+        verifyNoInteractions(recordItemListener);
+        verify(recordStreamFileListener).onError();
     }
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/RecordFileParserTest.java]: Can you add a test for when processed hash is empty in db? Ops does this quite a bit to reset environments.

**Comment by apeksharma** [comment]
looking at `@CachePut`.

**Comment by apeksharma** [comment]
Added @CachePut.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java [hunk 2]
```diff
@@ -139,125 +138,105 @@ public static String readPrevFileHash(String fileName) {
...
-                        case FileDelimiter.RECORD_TYPE_PREV_HASH:
-                            byte[] readFileHash = new byte[48];
-                            dis.read(readFileHash);
-
-                            if (Utility.hashIsEmpty(expectedPrevFileHash)) {
-                                log.error("Previous file hash not available");
-                                expectedPrevFileHash = Hex.encodeHexString(readFileHash);
+                byte typeDelimiter = dis.readByte();
+
+                switch (typeDelimiter) {
```
[the remark this answers is not in the document]
Comment (apeksharma) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java]: yes, and share them, i think this piece of code is partially duplicated in some other place too.
Tried not doing those in this PR though - keep changes small for faster easier reviews & quick progress (although that doesn't seem to be working).
